### ISSUE TITLE
digital_ocean module: got msg: dopy >= 0.2.3 required for this module all time

### ISSUE THREAD

**Issue by puliaiev** [ansible/ansible-modules-core#360]
##### Issue Type:

Bug Report
##### Ansible Version:

ansible 1.7.2
##### Environment:

OS X Yosemite
Version: 10.10.1
##### Summary:

All time got error: 
msg: dopy >= 0.2.3 required for this module all time

> pip list
> ansible (1.7.2)
> dopy (0.3.0)
##### Steps To Reproduce:

---
- hosts: local
  connection: local
  tasks:
  - digital_ocean: >
    state=present
    command=ssh
    name=digitalocean_rsa
    ssh_pub_key='ssh-rsa AAAAB3NzaC1yc'
    client_id={{client_id}}
    api_key={{api_key}}
##### Expected Results:

Expected Ok result
##### Actual Results:

failed: [localhost] => {"failed": true}
msg: dopy >= 0.2.3 required for this module

FATAL: all hosts have already failed -- aborting

PLAY RECAP *******************************************************************\* 
           to retry, use: --limit @/Users/sevaua/do.yaml.retry

localhost                  : ok=1    changed=0    unreachable=0    failed=1

**Comment by bcoca** [comment]
# Possible Misunderstanding

Hi!

Thanks very much for your submission to Ansible.  It sincerely means a lot to us.

We believe the ticket you have filed is being somewhat misunderstood, as one thing works a little differently than stated.

The digital ocean modules have extra an requirement, the 'dopy' library of at least version 0.2.3, as the 
error message states. You need to install that on your 'master'  to use the modules.

In the future, this might be a topic more well suited for the user list, which you can also post here if you'd like some more help with the above.
- https://groups.google.com/forum/#!forum/ansible-project

Thank you once again for this and your interest in Ansible!

**Comment by pmackay** [comment]
Why was this closed @bcoca? The report says that `pip list` shows `dopy` is present, but the module still reports it needs the library. I'm finding a similar situation. How to resolve?

**Comment by puliaiev** [comment]
I have added next line in hosts file:

localhost ansible_connection=local **ansible_python_interpreter=python**

https://groups.google.com/forum/?hl=ru#!topic/ansible-project/gjQM-rArtTg

**Comment by sjelfull** [comment]
That was helpful, @SevaUA.

**Comment by Paxa** [comment]
Got same error with ansible 1.9.4

solved by installing exact version:

```
sudo pip install 'dopy>=0.2.3,<=0.2.3'
```

The original error message is (without try-except):

```
>>> import dopy
>>> from dopy.manager import DoError, DoManager
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Python/2.7/site-packages/dopy/manager.py", line 10, in <module>
    from six import wraps
ImportError: cannot import name wraps
```

**Comment by lcacciagioni** [comment]
It didn't solve the issue for me, this is still valid please reopen this issue.

**Comment by abadger** [comment]
Please open a new issue (or post to the mailing list first)  The original
issue looks to have been a problem with how the environment was setup
rather than ansible code.

A new issue (or mailing list post) should include the version of ansible,
the output when you get the error, the version of dopy.

New comments on Closed issues often don't get noticed so commenting here
will only get attention if a developer Jake's across it by accident.

**Comment by lcacciagioni** [comment]
Fixed it seems to be that it needs `six` installed but this is not included explicitly in the dependencies, after installing it it works perfectly.

**Comment by Freika** [comment]
Same issue here:

```
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "msg": "dopy >= 0.3.2 required for this module"}
```

Installing exact version didn't help (I guess, it's because there is no 0.3.2 version, only 0.3.3)

**Comment by youfoundbruno** [comment]
I had a similar problem using my Macbook. The problem seemed to have origin within the "setuptools" python module that came pre-installed with my Mac. The solution for me was to upgrade that module to the last version, by running in the following in the command line: `pip install --upgrade setuptools --user python`
I hope this can be useful to someone.

**Comment by Freika** [comment]
I solved my problem by removing brew version of ansible and installing pip version.

**Comment by potterhe** [comment]
I solved by "/usr/local/Cellar/ansible/2.2.1.0_1/libexec/bin/pip install dopy"

**Comment by mjmare** [comment]
Uninstalling the brew installed ansible did not love this problem. Only after adding "ansible_python_interpreter=python" to the hosts.ini things worked.
```
mylocalhost ansible_connection=local ansible_python_interpreter=python
```
ansible and dopy (0.3.7) are installed into a virtualenv. Ansible-playbook is run from the virtualenv:
```
venv/bin/ansible-playbook
```

**Comment by jdickey** [comment]
Running into the same problems as well:

```
$ uname -v
Darwin Kernel Version 15.6.0: Sun Jun  4 21:43:07 PDT 2017; root:xnu-3248.70.3~1/RELEASE_X86_64
$ ansible --version
ansible 2.3.1.0
  config file = 
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Jul 18 2017, 09:16:53) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
$ python --version
﻿﻿﻿Python 2.7.10
$ /usr/local/Cellar/ansible/2.3.1.0/libexec/bin/python --version
Python 2.7.13
$
$
$
$ ansible-playbook digitalocean.yml                     

PLAY [digitalocean] ****************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] *************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [ensure ssh key exists] *******************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [ensure key exists at DigitalOcean] *******************************************************************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "msg": "dopy >= 0.3.2 required for this module"}
	to retry, use: --limit @/Users/jeffdickey/src/script/grid-demo/digitalocean.retry

PLAY RECAP *************************************************************************************************************************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=1   

$
```

The playbook (from [this DO tutorial](https://www.digitalocean.com/community/tutorials/how-to-use-the-digitalocean-api-v2-with-ansible-2-0-on-ubuntu-16-04)) is:

```
---
- hosts: digitalocean

  vars:
    do_token: fe6736... # elided for a public post

  tasks:

  - name: ensure ssh key exists
    user: >
      name={{ ansible_user_id }}
      generate_ssh_key=yes
      ssh_key_file=.ssh/id_rsa

  - name: ensure key exists at DigitalOcean
    digital_ocean: >
      state=present
      command=ssh
      name=my_ssh_key
      ssh_pub_key={{ lookup('file', '~/.ssh/id_rsa.pub') }}
      api_token={{ do_token }}
    register: my_ssh_key

  - name: ensure droplet one exists
    digital_ocean: >
      state=present
      command=droplet
      name=droplet-one
      size_id=512mb
      region_id=sgp1
      image_id=ubuntu-16-04-x64
      ssh_key_ids={{ my_ssh_key.ssh_key.id }}
      api_token={{ do_token }}
    register: droplet_one

  - debug: msg="IP is {{ droplet_one.droplet.ip_address }}"

```

Just to be utterly pedantic, I do not have an existing Droplet by the name of `droplet-one`:

```
$ doclt droplets list --no-color
┌──────────┬───────────────┬─────────────────┬────────┐
│ ID       │ Name          │ IPv4            │ Status │
├──────────┼───────────────┼─────────────────┼────────┤
│ xxxxxxxx │ suirdemo1-off │ 10.130.10.xxx   │ off    │
│          │               │ 128.199.105.xxx │        │
├──────────┼───────────────┼─────────────────┼────────┤
│ xxxxxxxx │ suirdemo2     │ 10.130.19.xxx   │ active │
│          │               │ 128.199.73.xxx  │        │
├──────────┼───────────────┼─────────────────┼────────┤
│ xxxxxxxx │ demo2         │ 10.130.39.xxx   │ active │
│          │               │ 139.59.115.xxx  │        │
├──────────┼───────────────┼─────────────────┼────────┤
│ xxxxxxxx │ demo1-off     │ 10.130.34.xx    │ off    │
│          │               │ 128.199.239.xxx │        │
└──────────┴───────────────┴─────────────────┴────────┘
$
```

Any ideas for how to proceed would be *greatly* appreciated.
